app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -280,31 +308,37 @@ object AppRepository {
         logging.d(javaClass.simpleName, "Got ${sessions.size} rows.")
 
         val highlights = readHighlights()
+        val highlightsAwareSessions = mutableListOf<SessionDatabaseModel>()
         for (highlight in highlights) {
             logging.d(javaClass.simpleName, "$highlight")
             for (session in sessions) {
-                if (session.sessionId == "${highlight.sessionId}") {
-                    session.highlight = highlight.isHighlight
+                highlightsAwareSessions += if (session.sessionId == "${highlight.sessionId}") {
+                    session.copy(isHighlight = highlight.isHighlight)
+                } else {
+                    session
                 }
             }
         }
```

Comment:
This is not doing the right thing. This code can add the same session multiple times.

If you're touching this anyway, please get rid of the nested loops. You could use the same approach that is used for alarms (return a set of session ids instead of creating temporary `Highlight` objects).

Reply:
:heavy_check_mark: Changed in 44371a6cb8ed6ce84e6864107035335da5931720. Please check if this is what you had in mind. I also adapted a related test.

Reply:
> Please check if this is what you had in mind.

:+1: